dataprofiler/tests/data_readers/test_csv_data.py

```diff
@@ -309,6 +429,60 @@ def test_set_header(self):
         self.assertEqual(1, csv_data.header)
         self.assertEqual('1', first_value)
 
+    def test_set_header_as_streams(self):
+        test_dir = os.path.join(test_root_path, 'data')
+        fp = 'csv/sparse-first-and-last-column-two-headers.txt'
+
+        for filename in [BytesIO(open(os.path.join(test_dir, fp), 'rb').read()),\
+                         StringIO(open(os.path.join(test_dir, fp), 'r').read())]:
+            # set bad header setting
+            options = dict(header=-2)
+            with self.assertRaisesRegex(ValueError,
+                                        '`header` must be one of following: auto, '
+                                        'none for no header, or a non-negative '
+                                        'integer for the row that represents the '
+                                        'header \(0 based index\)'):
+                csv_data = CSVData(filename, options=options)
+                first_value = csv_data.data.loc[0][0]
+
+            # set bad header setting
+            options = dict(header='abcdef')
+            with self.assertRaisesRegex(ValueError,
+                                        '`header` must be one of following: auto, '
+                                        'none for no header, or a non-negative '
+                                        'integer for the row that represents the '
+                                        'header \(0 based index\)'):
+                csv_data = CSVData(filename, options=options)
+                first_value = csv_data.data.loc[0][0]
```

Comment:
We can remove this as it is covered in the other tests. Additionally, if we did want these tests, I would suggest not putting them in the for loop as we would only need to test it once.